Warning on SYS_LOG_ERR instruction in spi_ll_stm32.c

Following warning is generated when running `tests/drivers/spi/spi_loopback`:
```
/local/mcu/zephyr/zephyr-project/drivers/spi/spi_ll_stm32.c:56:10: warning: format ‘%d’ expects argument of type ‘int’, but argument 7 has type ‘long unsigned int’ [-Wformat=]
   SYS_LOG_ERR("%s: err=%d", __func__,
          ^~~~~~~~~~~~~~~
```
This prevents CI to pass, for instance here:
https://github.com/zephyrproject-rtos/zephyr/pull/5541

--- a/drivers/spi/spi_ll_stm32.c
+++ b/drivers/spi/spi_ll_stm32.c
@@ -54,7 +54,7 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 
 	if (sr & SPI_STM32_ERR_MSK) {
 		SYS_LOG_ERR("%s: err=%d", __func__,
-			    sr & SPI_STM32_ERR_MSK);
+			    sr & (u32_t)SPI_STM32_ERR_MSK);
 
 		/* OVR error must be explicitly cleared */
 		if (LL_SPI_IsActiveFlag_OVR(spi)) {

